File: 504-base-7/base-7.java
class Solution {
    public String convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }
        StringBuilder res = new StringBuilder();
        boolean isNegative = (num < 0);
        num = Math.abs(num);

        while (num > 0) {
            res.append("" + (num % 7));
            num /= 7;
        }

        String ret = res.reverse().toString();
        return isNegative ? '-' + ret : ret;
    }
}

File: 504-base-7/base-7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return '0'
        ans = []
        isNegative = num < 0
        num = abs(num)

        while num:
            ans.append(str(num % 7))
            num //= 7

        result = "".join(ans[::-1])
        return '-' + result if isNegative else result